### PR TITLE
Fix the projection of tippyName when using ng-template 

### PR DESCRIPTION
**Pull request by onaluf**
This also adds [tippyContext] to pass a custom context to the template which solves #43

### CODE REVIEW

**Comment by onaluf** [comment]
I hope this works for you, don't hesitate if there is any changes that you want me to make to this! I've also added a unit test that passes but on my box some other test where failing on `master`.

File: projects/ngx-tippy-wrapper/src/lib/ngx-tippy.directive.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export class NgxTippyDirective implements OnInit, OnDestroy {
   @Input() tippyProps?: NgxTippyProps;
   @Input() tippyName?: string;
   @Input() tippyClassName?: string;
+  @Input() tippyContext: any;
```
Comment (farengeyt451) [inline projects/ngx-tippy-wrapper/src/lib/ngx-tippy.directive.ts:28]: `tippyContext` should be optional like previous attributes and I think it also should have some generic interface

File: projects/ngx-tippy-wrapper/src/lib/services/ngx-view.service.ts [hunk 2]
```diff
@@ -8,14 +8,13 @@ export class NgxViewService {
 
   constructor(private appRef: ApplicationRef) {}
 
-  getViewRefInstance(content: NgxTippyTemplate, tippyName?: string): ViewRef {
+  getViewRefInstance(content: NgxTippyTemplate, tippyName?: string, tippyContext?: any): ViewRef {
```
Comment (farengeyt451) [inline projects/ngx-tippy-wrapper/src/lib/services/ngx-view.service.ts:11]: Same here, pls don't use `any` type

File: projects/ngx-tippy-wrapper/src/lib/ngx-tippy.directive.ts
Comment (farengeyt451) [inline projects/ngx-tippy-wrapper/src/lib/ngx-tippy.directive.ts:1]: For `tippyContext` attribute we need definitely try to handle case when `context` could be changed dynamically as in other attributes `(tippyName, ngxTippy, tippyProps, tippyClassName)`.

As example we can take such primitvie case:

```html
<button
  ngxTippy
  tippyName="ng-template-ref-plus-context"
  [ngxTippy]="tplctx"
  [tippyProps]="ngTemplateRef"
  [tippyContext]="templateContext"
>
  Element with tooltip
</button>

<ng-template
  #tplctx
  let-name
  let-foo="foo"
>
  <div class="t-template">
    <span class="t-template__name">Passed tippyName <b>{{ name }}</b> and context <b>{{ foo }}</b></span>
  </div>
</ng-template>
```

```ts
  templateContext!: Record<string, any>;

  ngOnInit() {
    this.templateContext = { foo: 'Hello' };

    setTimeout(() => {
      this.templateContext = { foo: 'World' };
    }, 1000);
  }
```

Need to handle this dynamic `context` changes

File: projects/ngx-tippy-demo/src/app/app.component.html
Comment (farengeyt451) [inline projects/ngx-tippy-demo/src/app/app.component.html:1]: As example we can use somthing simpler like:

```html
<button
  ngxTippy
  tippyName="ng-template-ref-plus-context"
  [ngxTippy]="tplctx"
  [tippyProps]="ngTemplateRef"
  [tippyContext]="templateContext"
>
  Element with tooltip
</button>

<ng-template
  #tplctx
  let-name
  let-foo="foo"
>
  <div class="t-template">
    <span class="t-template__name">Passed tippyName <b>{{ name }}</b> and context <b>{{ foo }}</b></span>
  </div>
</ng-template>
```

File: README.md [hunk 3]
```diff
@@ -150,13 +150,16 @@ export class DemoComponent implements OnInit {
 | `tippyProps`     | NgxTippyProps | [tippyProps]="{ arrow: false, placement: 'bottom' }"                               |
 | `tippyName`      | string        | tippyName="awesomeName"                                                            |
 | `tippyClassName` | string        | tippyClassName="new-class" <br> _or_ <br> tippyClassName="new-class another-class" |
+| `tippyContext`   | any           | [tippyContext]="{ aKey: 'something' }" |
```
Comment (farengeyt451) [inline README.md:153]: Better to provide interface instead of `any`

**Review by farengeyt451 — CHANGES_REQUESTED** [review]
Thank you for your MR)

**Comment by onaluf** [comment]
Hope this is to you liking, don't hesitate if you want other changes! I tried adding a unit test for change detection in the context but didn't manage to.

**Pull request merged**